### PR TITLE
Add Portfolio Optimizer application

### PR DESCRIPTION
**Pull request by philippjfr**

### CODE REVIEW

**Comment by jbednar** [comment]
Looks good!

**Comment by philippjfr** [comment]
Not really tbh, needs a lot of cleanup but wanted to meet the deadline. Which I hear is Friday now?

**Comment by jsignell** [comment]
You can see the built version on the dev site: https://pyviz-dev.github.io/examples/portfolio_optimizer/portfolio.html

File: portfolio_optimizer/portfolio.ipynb [hunk 1]
```diff
@@ -93,6 +78,8 @@
     "4. Click on the Return/Volatility plot to select the desired risk/reward profile\n",
     "\"\"\"\n",
     "\n",
+    "# This will go away with panel 0.6.3\n",
```
Comment (philippjfr) [inline portfolio_optimizer/portfolio.ipynb]: Turns out this isn't actually fixed properly until 0.6.4

**Comment by jbednar** [comment]
@philippjfr , shouldn't this project include the original notebook that has all the human-readable discussion, and move the current .ipynb's file into a .py file that solely implements the dashboard version?  The project would then show the full dashboard when it's visited, but can then have a link to the standalone dashboard version.

**Comment by philippjfr** [comment]
>The project would then show the full dashboard when it's visited, but can then have a link to the standalone dashboard version.

Presumably you mean"The project would then show the notebook when it's visited"...?

The original notebook is quite different from the dashboard, which is why I didn't do this originally.

**Comment by jbednar** [comment]
Right, I meant show the notebook, not the dashboard, when first visited.  Even if it's sparse, it at least has _some_ verbiage directed at humans.  If you'd rather put such info into the current .ipynb file, be my guest, but I think there needs to be *something* readable...

File: portfolio_optimizer/portfolio.ipynb [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    "\n",
+    "To optimize your portfolio:\n",
+    "\n",
+    "1. Upload a CSV of the daily stock time series for the stocks you are considering\n",
+    "2. Select the stocks to be included.\n",
+    "3. Run the Analysis\n",
+    "4. Click on the Return/Volatility plot to select the desired risk/reward profile\n",
+    "\"\"\"\n",
+    "\n",
+    "# This will go away with panel 0.6.3\n",
```
Comment (philippjfr) [inline portfolio_optimizer/portfolio.ipynb]:
```suggestion
```

File: portfolio_optimizer/portfolio.ipynb [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+    "pn.widgets.FileInput._rename = {'filename': None, 'name': None}\n",
```
Comment (philippjfr) [inline portfolio_optimizer/portfolio.ipynb]:
```suggestion
```

File: portfolio_optimizer/portfolio.ipynb [hunk 4]
```diff
@@ -0,0 +1,566 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import hvplot.pandas"
```
Comment (jsignell) [inline portfolio_optimizer/portfolio.ipynb]:
```suggestion
    "import hvplot.pandas  # noqa"
```

This'll fix the flake error

File: portfolio_optimizer/anaconda-project.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+channels: []
+
+packages: &pkgs
+- python=3.6
+- notebook=5.7.8
+- ipykernel=5.1.0
+- nomkl
+- bokeh=1.3.4
+- holoviews=1.12.3
```
Comment (jsignell) [inline portfolio_optimizer/anaconda-project.yml]:
```suggestion
- holoviews=1.12.3
- hvplot=0.4.0
```

**Pull request merged**